polkadot/runtime/rococo/src/lib.rs

```diff
@@ -1532,8 +1534,8 @@ pub type Block = generic::Block<Header, UncheckedExtrinsic>;
 pub type SignedBlock = generic::SignedBlock<Block>;
 /// `BlockId` type as expected by this runtime.
 pub type BlockId = generic::BlockId<Block>;
-/// The `SignedExtension` to the basic transaction logic.
-pub type SignedExtra = (
+/// The extension to the basic transaction logic.
+pub type TxExtension = (
```

Comment:
I kind of think the order should be changed to: having at the end:`CheckMetadataHash` then `ChargeTransactionPayment` and then `CheckWeight`.

This is because `CheckWeight` doesn't refund so it can be after `ChargeTransactionPayment`, and it needs the weight of `ChargeTransactionPayment` so it must be after it.

Reply:
Yes, in the final version the order would have to be `(..., TxFeePaymentExtension, CheckWeight)`. Reverse could work, but the extension used to pay the tx fees (could be `ChargeAssetTxPayment` for example) can have a more complex weight. This combined with the idea to assume worst case scenario for the post dispatch weight and subtract the unspent would enable us to correctly refund the fee if the tx fee payment extension is before `CheckWeight` in the pipeline and we know that `CheckWeight` has a constant weight, which we could account for beforehand. Afterwards, `CheckWeight` correctly updates the block weight.

Reply:
Let's rearrange all of the extension pipelines in our repo in a follow up to avoid adding the diff and the test changes here.